### DISCUSSION TITLE
Modular Weekdays

### DISCUSSION THREAD

**Discussion by aqllmcdavid** [vkurko/calendar#84] [Ideas]
Is there a way to make the calendar week day display work with modular events that do not factor in dates?
If I wanted to drag and drop events that happen from a standard Monday to Friday 9am-5pm it is a simple enough procedure and the dates could be stripped at the backend, but if I wanted to drag an event that will for example span from Friday 5pm, through Saturday and Sunday to Monday 9am, is this even possible or feasible to do this action and display it (without switching to the next week) with this package?

**Comment by vkurko** [comment]
I'm not entirely sure that I understood your question correctly, so please correct me if this is the case.

To display a different span of days on the weekly calendar (not Mon-Fri), you can use the [firstDay](https://github.com/vkurko/calendar#firstday) setting, which can be anything from Sun to Sat.

As for dragging events, it is not currently possible to completely drag an event from one week to another. But it is possible to drag the event so that its beginning is at the very end of the displayed week, then the end of the event will naturally move to the next week.

**Comment by aqllmcdavid** [comment]
Thank you for your reply, my issue is not regarding the start of the week, but the modular way in which weeks run. Perhaps I am looking at the wrong kind of information display because a calendar date is linear, whereas I am looking for something that would indefinitely wrap around itself.
So if for example you add 2 minutes to a clock when the minute is at 59 and ignore the hour, you would want it to show 1 as opposed to 61 and I am looking to do the same with weeks.
I know this is probably fairly niche issue, perhaps I am looking at the wrong library to do this but I was just wondering if it was possible.

**Comment by vkurko** [comment]
Agree on the niche issue. Perhaps an example from real life would bring more clarity to your need.

**Comment by aqllmcdavid** [comment]
![day-wrap](https://user-images.githubusercontent.com/101651862/219686902-4be854ad-b806-4332-82ed-4423339a9756.png)

A week display without dates, where the end can appear "behind" of the start due to the modular nature of a week.

**Reply by vkurko** [reply]
> where the end can appear "behind" of the start

Unfortunately, this is not possible in the current version of the calendar.
